### DISCUSSION TITLE
Custom loader in table row

### DISCUSSION THREAD

**Discussion by haresh-samnani** [rsuite/rsuite-table#427] [Q&A] [answered]
Hi,

I am currently using the latest version of rsuite table. I know that I can show loader for the whole table data using prop ```loading```. I am planning to use ```react-skeleton-loading``` to show skeleton effect in each table row cell but I don't know how to because table accepts data (generally my API response array of objects) in the ```data``` prop.

Can anybody help in achieving this?

**Comment by simonguo** [comment, marked as answer]
Hi @haresh-samnani you can use rsuite's `Placeholder` component.

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/1203827/235816274-d186ad82-9c88-4fd7-9fa1-f94b13447f17.png">

> https://table.rsuitejs.com/#loading

## Usage

```tsx
import { Placeholder, Table } from 'rsuite';

const renderLoading = () => {
  return (
    <div
      style={{
        position: 'absolute',
        width: '100%',
        height: '100%',
        background: '#fff',
        padding: 20,
        zIndex: 1
      }}
    >
      <Placeholder.Grid rows={20} columns={6} active />
    </div>
  );
};

const App = () => {
  return (
    <Table loading renderLoading={renderLoading}>
      ...
    </Table>
  );
};

```

**Reply by haresh-samnani** [reply]
@simonguo Thank you so much for the reply. I will be implementing this and mark it as an answer once done.

**Comment by haresh-samnani** [comment]
Thank you. Marked as an answer!
